sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java

```diff
@@ -163,29 +204,66 @@
             EncryptionItemRequestOptions.class);
 
-        if (encryptionItemRequestOptions != null && encryptionItemRequestOptions.getEncryptionOptions() != null) {
-            Preconditions.checkArgument(partitionKey != null, "partitionKey cannot be null for operations using "
-                + "EncryptionContainer.");
+        if (encryptionItemRequestOptions == null || encryptionItemRequestOptions.getEncryptionOptions() == null) {
+            return container.createItem(item, partitionKey, requestOptions);
+        }
+
+
+        Preconditions.checkArgument(partitionKey != null, "partitionKey cannot be null for operations using "
+            + "EncryptionContainer.");
+
+
+        EncryptableItem encryptableItem = Utils.as(item, EncryptableItem.class);
+        if (encryptableItem != null) {
+            byte[] streamPayload = encryptableItem.toStream(this.getItemDeserializer());
+
+            Mono<CosmosItemResponse<byte[]>> rspMono = this.createItemHelper(streamPayload,
+                partitionKey,
+                requestOptions,
+                false);
+
 
-            return Mono.defer(() -> {
-                byte[] payload = cosmosSerializerToStream(item);
-                Mono<CosmosItemResponse<byte[]>> result = this.createItemStream(payload, partitionKey,
-                    encryptionItemRequestOptions);
+            return rspMono.map(
+                rsp -> {
+                    encryptableItem.setDecryptableItem(
+                        // EncryptionProcessor.BaseSerializer.FromStream<JObject>(responseMessage.Content),
+                        getItemDeserializer().parseFrom(ObjectNode.class,
+                            EncryptionModelBridgeInternal.getByteArrayContent(rsp)),
+                        this.encryptor,
+                        this.getItemDeserializer()
+                    );
 
-                return result.map(rsp -> (CosmosItemResponse<T>) this.responseFactory.createItemResponse(rsp,
-                    item.getClass()));
+                    return createEncryptionItemResponse(rsp, (T) encryptableItem);
+                }
+            ).subscribeOn(encryptionScheduler);
 
-            }).subscribeOn(encryptionScheduler);
         } else {
-            return container.createItem(item, partitionKey, requestOptions);
+            byte[] streamPayload = cosmosSerializerToStream(item);
+            Mono<CosmosItemResponse<byte[]>> response = createItemHelper(
+                streamPayload,
+                partitionKey,
+                requestOptions,
+                true
+            );
+
+            return response
+                .publishOn(encryptionScheduler)
+                .map(
+                    rsp -> {
+                        return this.responseFactory.createItemResponse(rsp, (Class<T>) item.getClass());
+                    }
+                );
+
+            // TODO: no support for streaming
```

Comment:
???

Reply:
we don't have support for streaming anywhere in the SDK.
I am considering wiring up some byte[] API. 

this TODO is only a hint to myself no actual work is remaining.
I will reword it to make it cleaner.